pkg/pool-weighted/contracts/managed/ManagedPool.sol

```diff
@@ -277,58 +285,101 @@ contract ManagedPool is ManagedPoolSettings {
         uint256 balanceTokenIn,
         uint256 balanceTokenOut,
         bytes32 poolState
-    ) internal view returns (uint256) {
-        uint256 tokenInWeight;
-        uint256 tokenOutWeight;
-        uint256 scalingFactorTokenIn;
-        uint256 scalingFactorTokenOut;
-        uint256 swapFeeComplement;
-        {
-            uint256 weightChangeProgress = ManagedPoolStorageLib.getGradualWeightChangeProgress(poolState);
-            (tokenInWeight, scalingFactorTokenIn) = _getTokenInfo(request.tokenIn, weightChangeProgress);
-            (tokenOutWeight, scalingFactorTokenOut) = _getTokenInfo(request.tokenOut, weightChangeProgress);
+    ) internal view returns (uint256 amountCalculated) {
+        uint256 swapFeeComplement = ManagedPoolStorageLib.getSwapFeePercentage(poolState).complement();
+        SwapTokenData memory swapTokenData = _getSwapTokenData(request, poolState);
 
-            swapFeeComplement = ManagedPoolStorageLib.getSwapFeePercentage(poolState).complement();
-        }
+        balanceTokenIn = _upscale(balanceTokenIn, swapTokenData.scalingFactorTokenIn);
+        balanceTokenOut = _upscale(balanceTokenOut, swapTokenData.scalingFactorTokenOut);
 
-        balanceTokenIn = _upscale(balanceTokenIn, scalingFactorTokenIn);
-        balanceTokenOut = _upscale(balanceTokenOut, scalingFactorTokenOut);
+        uint256 endingBalanceTokenIn;
+        uint256 endingBalanceTokenOut;
```

Comment:
These are never set

Reply:
Since the block below sets `amountCalculated` (and we know amount given - `request.amount`), we could pass both to the `checkBreaker` function I suggest and adjust theb alances with something like the following

```
    function _getAmounts(
        SwapKind kind,
        uint256 amountGiven,
        uint256 amountCalculated
    ) private pure returns (uint256 amountIn, uint256 amountOut) {
        if (kind == SwapKind.GIVEN_IN) {
            (amountIn, amountOut) = (amountGiven, amountCalculated);
        } else {
            // SwapKind.GIVEN_OUT
            (amountIn, amountOut) = (amountCalculated, amountGiven);
        }
    }
```

and then do

```
balance token in += amount in
balance token put -= amount out
```